packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -954,9 +966,17 @@ function readFromUnsubcribedMutableSource<Source, Snapshot>(
     // but there's nothing we can do about that (short of throwing here and refusing to continue the render).
     markSourceAsDirty(source);
 
+    if (__DEV__) {
+      if (mutableSourceSideEffectDetected) {
+        console.warn(
+          'React detected an unsafe side effect during render. Mutable sources should not be modified during render.',
```

Comment:
I'm not feeling particularly clever with this warning. We could probably do better. Suggestions?

Reply:
"unsafe side effect" sounds a bit vague to me, like how can React possibly know that I ran a side effect? I would maybe say that `A mutable source was mutated while the %s component was rendering. This is not supported. Move any mutations into event handlers or effects.`

Reply:
Sounds much better :)